test/unit-tests/otcWrapperV2.test.ts

```diff
@@ -0,0 +1,2038 @@
+      const premium = parseUnits('5000', 6)
+      const collateralAmount = parseUnits('30000', 6)
+      const orderID = 1
+
+      // USDC depegs to 0
+      await oracle.setRealTimePrice(usdc.address, 0)
+      await oracle.setRealTimePrice(wbtc.address, parseUnits('20000', 8))
+
+      await wbtc.mint(marketMaker, collateralAmount)
+      await wbtc.approve(otcWrapperProxy.address, collateralAmount, { from: marketMaker })
+
+      await expectRevert(
+        otcWrapperProxy.executeOrder(
+          orderID,
+          userSignature1,
+          mmSignatureUSDC1,
+          premium,
+          wbtc.address,
+          collateralAmount,
+          {
+            from: marketMaker,
+          },
+        ),
+        'OTCWrapper: invalid USDC price',
+      )
+
+      // USDC repegs
+      await oracle.setRealTimePrice(usdc.address, scaleBigNum(1, 8))
+    })
+    it('fee is properly calculated in a depeg scenario', async () => {
+      // user places order
+      const strikePrice = scaleBigNum(1300, 8)
+      const size = parseUnits('100', 8)
+
+      /*       await otcWrapperProxy.placeOrder(weth.address, false, strikePrice, expiry, parseUnits('5000', 6), size, {
+        from: user,
+      })
+ */
+      // order gets executed
+      const premium = parseUnits('5000', 6)
+      const collateralAmount = parseUnits('30000', 6)
+      const orderID = 1
+
+      // USDC depegs 50%
+      await oracle.setRealTimePrice(usdc.address, scaleBigNum(5, 7))
+      const originalOrderFee = parseUnits('1500', 6) // 1% of notional 150k notional
+      const depegOrderFee = originalOrderFee.mul(2)
+
+      const otcWrapperUSDCBalBefore = new BigNumber(await usdc.balanceOf(otcWrapperProxy.address))
+
+      await otcWrapperProxy.executeOrder(
+        orderID,
+        userSignature1,
+        mmSignatureUSDC1,
+        premium,
+        usdc.address,
+        collateralAmount,
+        {
+          from: marketMaker,
+        },
+      )
+
+      const otcWrapperUSDCBalAfter = new BigNumber(await usdc.balanceOf(otcWrapperProxy.address))
+
+      assert.equal(otcWrapperUSDCBalAfter.minus(otcWrapperUSDCBalBefore).toString(), depegOrderFee.toString())
+
+      // USDC repegs
+      await oracle.setRealTimePrice(usdc.address, scaleBigNum(1, 8))
+    })
+  })
+
+  describe('claimFees', () => {
+    it('successfully claims fees', async () => {
+      const amountToClaim = parseUnits('3000', 6)
+
+      assert.equal((await usdc.balanceOf(beneficiary)).toString(), '0')
+      assert.isAbove(Number(await usdc.balanceOf(otcWrapperProxy.address)), 0)
+
+      const tx = await otcWrapperProxy.claimFees()
+
+      assert.equal((await usdc.balanceOf(beneficiary)).toString(), amountToClaim.toString())
+      assert.equal((await usdc.balanceOf(otcWrapperProxy.address)).toString(), '0')
+
+      const { logs } = tx
+      assert.equal(logs[0].args.amountClaimed.toString(), amountToClaim)
+    })
+    it('reverts if contract balance is 0', async () => {
```

Comment:
good practice to have revert unit tests at beginning of function unit tests

Reply:
Was last due to it being a usdc (not wbtc) balanceOf test before - given claimFees change of adding asset. Moved it to the beginning now

https://github.com/ribbon-finance/GammaProtocol-OTC/pull/10/commits/351e62c62c691b83aa221af5fc301befbb5398ab